OpenRA.Mods.Common/Traits/Buildings/Building.cs

```diff
@@ -154,14 +154,17 @@ public Actor FindBaseProvider(World world, Player p, CPos topLeft)
 
 			foreach (var bp in world.ActorsWithTrait<BaseProvider>())
 			{
+				if (bp.Trait.IsTraitDisabled || bp.Trait.IsTraitPaused)
```

Comment:
This can be removed.  The `bp.Trait.Ready()` 3 lines down (should) already check both of these.